src/callbacks_step/save_restart_dg.jl

```diff
@@ -327,4 +327,53 @@ function load_restart_file_on_root(mesh::Union{ParallelTreeMesh, ParallelP4estMe
 
     return u_ode
 end
+
+function save_restart_controller(mesh, integrator,
+                                 controller, restart_callback)
+    if mpi_isroot()
+        save_restart_controller(integrator, controller, restart_callback)
+    end
+    if mpi_isparallel()
+        MPI.Barrier(mpi_comm())
+    end
```

Comment:
Why do you need a barrier here? If it is really necessary, it should be documented why.

Reply:
You're right, it's not necessary any more.